cmd/frontend/graphqlbackend/search_results.go

```diff
@@ -1122,3 +1127,17 @@ func regexpPatternMatchingExprsInOrder(patterns []string) string {
 	}
 	return "(" + strings.Join(patterns, ").*?(") + ")" // "?" makes it prefer shorter matches
 }
+
+func validateRepoHasFileUsage(q *query.Query) error {
+	// Validate usage of `repohasfile` filter
+	rawQuery := q.Syntax.Input
+	if strings.Contains(rawQuery, "type:repo") && strings.Contains(rawQuery, "repohasfile:") {
+		return errors.New("repohasfile does not currently return repository results. Support for repository results is in progress. Subscribe to https://github.com/sourcegraph/sourcegraph/issues/4584 for updates")
```

Comment:
```suggestion
		return errors.New("repohasfile does not currently return repository results. Subscribe to https://github.com/sourcegraph/sourcegraph/issues/4584 for updates")
```

The status here "in progress" may not be true if we've e.g. finished it and a customer is on an older version of Sourcegraph, better to just link the issue I think